```
build$ make -j8 mcpelauncher-client
[  2%] Built target eglut
[  5%] Built target linux-gamepad
[  6%] Built target filepicker
[  3%] Built target logger
Cleaning libffi directory...
[  7%] Built target base64
[ 14%] Built target simpleipc
HEAD ist jetzt bei 20562ac Fix for AArch64. Release as 3.2.1.
[ 32%] Built target linker
[ 36%] Built target ffi
[ 40%] Built target gamewindow
[ 41%] Built target mcpelauncher-common
[ 51%] Built target libc-shim
[ 53%] Built target cll-telemetry
[ 54%] Built target file-util
[ 71%] Built target fake-jni
[ 75%] Built target mcpelauncher-core
[ 75%] Built target daemon-client-utils
[ 76%] Building CXX object baron/CMakeFiles/baron.dir/src/impl/interface/native/field.cpp.o
[ 77%] Built target msa-daemon-client
/home/christopher/ng/mcpelauncher-manifest/baron/src/impl/interface/native/field.cpp:65:3: error: use of undeclared identifier 'LOG_GET_FABRICATED_FIELD'
  DEFINE_ARBITRARY_CALLBACKS
  ^
/home/christopher/ng/mcpelauncher-manifest/baron/src/impl/interface/native/field.cpp:52:2: note: expanded from macro 'DEFINE_ARBITRARY_CALLBACKS'
 LOG_GET_FABRICATED_FIELD\
 ^
/home/christopher/ng/mcpelauncher-manifest/baron/src/impl/interface/native/field.cpp:79:3: error: use of undeclared identifier 'LOG_GET_FABRICATED_FIELD'
  DEFINE_ARBITRARY_CALLBACKS
  ^
/home/christopher/ng/mcpelauncher-manifest/baron/src/impl/interface/native/field.cpp:52:2: note: expanded from macro 'DEFINE_ARBITRARY_CALLBACKS'
 LOG_GET_FABRICATED_FIELD\
 ^
2 errors generated.
make[3]: *** [baron/CMakeFiles/baron.dir/build.make:180: baron/CMakeFiles/baron.dir/src/impl/interface/native/field.cpp.o] Fehler 1
make[2]: *** [CMakeFiles/Makefile2:1174: baron/CMakeFiles/baron.dir/all] Fehler 2
make[1]: *** [CMakeFiles/Makefile2:1280: mcpelauncher-client/CMakeFiles/mcpelauncher-client.dir/rule] Fehler 2
make: *** [Makefile:474: mcpelauncher-client] Fehler 2
```
Shure if you want to see it, clang-9.0.0

--- a/examples/src/main.cpp
+++ b/examples/src/main.cpp
@@ -12,39 +12,39 @@ public:
  DEFINE_CLASS_NAME("com/example/ExampleClass")
 
  JInt exampleField1;
- JString exampleField2;
+ std::shared_ptr<JString> exampleField2;
 
  //This constructor is visible to the JVM
  ExampleClass() :
   exampleField1(10),
-  exampleField2("Hello World!")
+  exampleField2(std::make_shared<JString>("Hello World!"))
  {}
 
  //This constructor is visible to the JVM
- ExampleClass(JInt, JString *) :
+ ExampleClass(JInt, std::shared_ptr<JString>) :
   exampleField1(0),
-  exampleField2{""}
+  exampleField2{std::make_shared<JString>("")}
  {}
 
  //This constructor is visible to the JVM
- ExampleClass(JDouble, ExampleClass *) :
+ ExampleClass(JDouble, std::shared_ptr<ExampleClass>) :
   exampleField1(0),
-  exampleField2{""}
+  exampleField2{std::make_shared<JString>("")}
  {}
 
  //This constructor is not visible to the JVM and thus no compile-time error is generated
  //for the illegal constructor prototype
  ExampleClass(JBooleanArray arr):
   exampleField1(-1),
-  exampleField2("This constructor is not registered on the JVM!")
+  exampleField2(std::make_shared<JString>("This constructor is not registered on the JVM!"))
  {}
 
  JInt exampleFunction() {
   return exampleField1;
  }
 
- JString* getMyString() {
-  return &exampleField2;
+ std::shared_ptr<JString> getMyString() {
+  return exampleField2;
  }
 
  inline static void exampleStaticFunction(JDouble d) {
@@ -75,8 +75,8 @@ BEGIN_NATIVE_DESCRIPTOR(ExampleClass)
  {Field<&ExampleClass::exampleField2> {}, "exampleField2"},
  //Register constructors
  {Constructor<ExampleClass> {}},
- {Constructor<ExampleClass, JInt, JString*> {}},
- {Constructor<ExampleClass, JDouble, ExampleClass *> {}}
+ {Constructor<ExampleClass, JInt, std::shared_ptr<JString>> {}},
+ {Constructor<ExampleClass, JDouble, std::shared_ptr<ExampleClass>> {}}
 END_NATIVE_DESCRIPTOR
 
 static void bruh(JNIEnv *env, jobject clazzOrInst, JDouble value) {

--- a/src/jni/native/array.cpp
+++ b/src/jni/native/array.cpp
@@ -65,7 +65,7 @@ for (JInt i = start; i < start + len; i++) {\
 
 namespace FakeJni {
  void* NativeInterface::getPrimitiveArrayCritical(jarray jarr, jboolean * copy) const {
-  auto arr = std::reinterpret_pointer_cast<JArray<void*, true>>(env.resolveReference(jarr));
+  auto arr = std::static_pointer_cast<JArray<void*, true>>(env.resolveReference(jarr));
   //We will never copy for now.
   if (copy != nullptr) {
       *copy = JNI_FALSE;

